handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1967,6 +1968,9 @@ public void handlerAdded(final ChannelHandlerContext ctx) throws Exception {
             // With TCP Fast Open, we write to the outbound buffer before the TCP connect is established.
             // The buffer will then be flushed as part of establishing the connection, saving us a round-trip.
             startHandshakeProcessing(active);
+            // If we weren't able to include client_hello in the TCP SYN (e.g. no token, disabled at the OS) we have to
+            // flush pending data in the outbound buffer later in channelActive().
+            needsFlush |= fastOpen && channel.unsafe().outboundBuffer().totalPendingWriteBytes() > 0;
```

Comment:
`channel.unsafe().outboundBuffer()` may return `null` when a channel was closed we should add a null check.

```suggestion
            ChannelOutboundBuffer outboundBuffer = channel.unsafe().outboundBuffer();
            needsFlush |= fastOpen && outboundBuffer != null && outboundBuffer.totalPendingWriteBytes() > 0;
```

Reply:
good catch! I will also flush if `outboundBuffer == null` (better to do an extra flush rather than miss a flush).